.github/workflows/tests.yaml

```diff
@@ -5,31 +5,38 @@ name: Lint & Test
 
 on:
   push:
-    branches: [main]
+    branches: [ main ]
   pull_request:
-    branches: [main]
+    branches: [ main ]
 
 jobs:
-  build:
-    runs-on: ubuntu-latest
+  test-lint:
     strategy:
+      fail-fast: false
       matrix:
-        python-version: [3.9]
+        python-version: [ 3.9.4 ]
+        poetry-version: [ 1.1.5 ]
```

Comment:
This seems strange. Why do we need a matrix strategy to test a single version?

Reply:
That is approach from Action example https://github.com/abatilo/actions-poetry
I've just used it as is. If it is terrible I can always fix it but to tell the truth I like to see all versions in one section that provides a fast glimpse at the project